client/src/utils/api.js

```diff
@@ -1,22 +1,43 @@
 // File to contain methods to return data from API calls
 import axios from "axios";
-
+import { BASE_URL } from "./constants";
 /**
  * Get internships with given search options. Returns a promise that resolves to
  * Internship[]
  *
  * @param {*} searchOptions - shape to be determined by backend
  * @return {Promise<Internship[]>}
  */
-export const getSearchInternships = (searchOptions) => {
-  const url = "";
-  const options = {};
-
-  return [
-    // Fill out mock data here
-  ];
+export const getSearchInternships = async (searchOptions) => {
+  const url = BASE_URL;
 
-  // return axios.get(url, options);
+  if (!searchOptions) {
+    const response = await axios
+      .get(`${url}/jobs`, {
+        headers: { "Content-Type": "application/json" },
+      })
+      .then(() => {
+        return response;
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  } else {
+    const response = await axios
+      .get(`${url}/jobs`, {
+        headers: { "Content-Type": "application/json" },
+        params: {
+          title: searchOptions["title"],
+          location: searchOptions["location"],
+        },
+      })
+      .then(() => {
+        return response;
+      })
+      .catch((error) => {
+        console.log(error);
+      });
```

Comment:
Async-await patterns we can wrap the async command in a try catch, cause the typical promise.then return is already handled by the await. So you should be able to just do `Const response = await axios.get()`